### PR TITLE
YR/Handle long running pipelines, and commits with no pipelines/CIAC-9386

### PR DESCRIPTION
**Pull request by RosenbergYehuda**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://jira-dc.paloaltonetworks.com/browse/CIAC-9386

## Description
This PR handles 2 edge cases:

- my current pipeline ended, but the previous one is still running.
- my commit had a pipeline, but the previous commit has no pipeline. (was merged within the same 5 minutes)

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Review by kobymeir — CHANGES_REQUESTED** [review]

File: Tests/scripts/common.py [hunk 1]
```diff
@@ -345,30 +361,124 @@ def get_slack_user_name(name: str | None, name_mapping_path: str) -> str:
...
+        commit_index: The index of the current commit.
+        list_of_commits: A list of commits.
+        list_of_pipelines: A list of pipelines.
+    Returns:
+
+    """
+    for index in reversed(range(1, commit_index)):
+        # the list of commits in in ascending order, newer commits are first
+        current_commit = list_of_commits[index - 1]
+        previous_commit = list_of_commits[index]
```
[the remark this answers is not in the document]
Comment (RosenbergYehuda) [reply Tests/scripts/common.py:446]: fixed

File: Tests/scripts/common.py [hunk 2]
```diff
@@ -345,30 +361,124 @@ def get_slack_user_name(name: str | None, name_mapping_path: str) -> str:
...
+    suspicious_commits = []
+    if direction == "newer":
+        # since the list of commits is in ascending order, to get the next commit we need to go backwards
+        for index in reversed(range(0, current_commit_index - 1)):
+            next_commit = list_of_commits[index]
+            suspicious_commits.append(list_of_commits[index + 1])
+            next_pipeline = get_pipeline_by_commit(next_commit, list_of_pipelines)
+            if next_pipeline:
+                return next_pipeline, suspicious_commits
+    elif direction == "older":
```
[the remark this answers is not in the document]
Comment (RosenbergYehuda) [reply Tests/scripts/common.py:476]: fixed

File: Tests/scripts/infrastructure_tests/common_test.py [hunk 3]
```diff
@@ -302,3 +302,133 @@ def test_get_slack_user_name__name_is_github_actions_bot():
     name = "github-actions[bot]"
     result = get_slack_user_name(name, str(Path(__file__).parent / 'tests_data/test_mapping.json'))
     assert result == "docker images bot owner"
+
+
+COMMITS = ['commit1', 'commit2', 'commit3', 'commit4', 'commit5']
+PIPLINES = ['pipeline1', 'pipeline2', 'pipeline3', 'pipeline4', 'pipeline5']
+
+
+def test_was_message_already_sent__was_sent_for_true_pivot(mocker, commits=COMMITS, pipelines=PIPLINES):
```
Comment (kobymeir) [inline Tests/scripts/infrastructure_tests/common_test.py:311]: why UT have a default value? bad practice.

File: Tests/scripts/infrastructure_tests/common_test.py [hunk 4]
```diff
@@ -302,3 +302,133 @@ def test_get_slack_user_name__name_is_github_actions_bot():
...
+    Then:
+        It should return True since the message was already sent for newer pipelines
+    """
+    mocker.patch('Tests.scripts.common.get_pipeline_by_commit', side_effect=lambda commit, pipelines: commit)
+    mocker.patch('Tests.scripts.common.is_pivot', return_value=True)
+
+    assert was_message_already_sent(2, commits, pipelines) is True
+
+
+def test_was_message_already_sent__was_sent_for_false_pivot(mocker, commits=COMMITS, pipelines=PIPLINES):
```
Comment (kobymeir) [inline Tests/scripts/infrastructure_tests/common_test.py:326]: same

**Review by kobymeir — CHANGES_REQUESTED** [review]

File: Tests/scripts/infrastructure_tests/common_test.py (hunk 3 above)
Comment (RosenbergYehuda) [reply Tests/scripts/infrastructure_tests/common_test.py:311]: Sorry, fixed

**Review by kobymeir — APPROVED** [review]
